app/public/js/localSprites.js

```diff
@@ -217,17 +253,88 @@ function addFireballContrailParticle(parent, decayFrames, parentPreScalingXSize,
-    //parent.contrailParticles.push(fireballContrailParticle);
-    localSprites.push(fireballContrailParticle); //BROKEN: Should push to parent.contrailParticles, but then render.js should render things in that array. I don't know the syntax for that yet, I don't think.
+    var fireballParticle = new SimpleSprite({frames}, randomX, randomY, 0, 0, 1.25, 2.5);
+    if (type === 0) fireballParticle.name = 'fireballContrailParticle';
+    if (type === 1) fireballParticle.name = 'fireballDetonationParticle';
+    fireballParticle.framesToLive = decayFrames;
+    fireballParticle.scaleOscillation = true;
+    fireballParticle.xScalePerFrame = fireballParticle.xScale / fireballParticle.framesToLive;
+    fireballParticle.yScalePerFrame = fireballParticle.yScale / fireballParticle.framesToLive;
+    fireballParticle.xScaleMin = 0;
+    fireballParticle.yScaleMin = 0;
+    fireballParticle.rotationPerFrame = 50;
+    //fireballParticle.msBetweenFrames = Math.round((decayFrames * 50 /*or framerate*/) / frames.length) + 1; //'+1' hopefully keeps the animation from starting to loop just before the pariticle dies.  //would be better to also have a continuous alpha fade happen during this time. Could also scale down if that weren't build into the animation frames already.
+    //parent.contrailParticles.push(fireballParticle);
+    if (fireballParticle.name === 'fireballContrailParticle') localSprites.push(fireballParticle); //WRONG: Should push to parent.contrailParticles, but then render.js should render things in that array. I don't know the syntax for that yet, I don't think.
+    if (fireballParticle.name === 'fireballDetonationParticle') {
+        var randomVX,
+        randomVY;
+        if (Math.random() < 0.5) {
+            randomVX = -(Math.random() * 50);
+        } else {
+            randomVX = Math.random() * 50;
+        }
+        if (Math.random() < 0.5) {
+            randomVY = -(Math.random() * 50);
+        } else {
+            randomVY = Math.random() * 50;
+        }
+        fireballParticle.vx = randomVX;     //BROKEN: I'm not seeing these detonation particles move how they should.
+        fireballParticle.vy = randomVY + 4;
+        localSprites.push(fireballParticle);
+    }
+}
+
+function addFireballDetonation(parent, numberOfFragments, parentPreScalingXSize, parentPreScalingYSize) {
+    var detonationParticles = [];
+    for (var i = 0; i < numberOfFragments; i++) {
+        var newParticle = addFireballParticle(parent, 30, parentPreScalingXSize, parentPreScalingYSize, 1);
+        detonationParticles.push(newParticle);
+    }
+    for (var j = 0; j < detonationParticles.length; j++) {
+
+    }
+}
+
+
+function addTriggerZone(left, top, width, height, target, spawnedObjectType, spawnedObjectXOffset, spawnedObjectYOffset, xForce, yForce, cooldownFrames) {
```

Comment:
It looks like maybe you want a generic `addTriggerZone` that accepts properties common to all triggers, and then maybe distinct triggers for specific type like `addSpawnTrigger` or `addForceTrigger`? That way you won't have to deal with as many arguments for each function.